app/packs/src/components/structureEditor/KetcherEditor.js

```diff
@@ -1,45 +1,845 @@
 /* eslint-disable react/require-default-props */
```

Comment:
File is too big: 
- suggestion: besides KetcherEditor functional component
  - break it into:
    - KetcherActions
    - Process/Manipulate KetcherData
    - KetcherValidations
    -  unit tests are missing for this functional component?